chore(storybook): Fix toolbar state not updated immediately after an editor state change

ayushanand18 opened the pull request:
<!-- Thank you for your contribution to the Typist repository. -->

<!-- Please remove all sections that are not relevant. -->

## Overview

<!-- Please include a summary of the changes and the related issue. -->
<!-- Please also include relevant motivation and context. -->
This PR tries to fix #259 by replacing the eventListener on Storybook.

### Changes Introduced
* `onSelectionUpdate` changed to `onTransation` on `typist editor toolbar decorator`.
    ```javascript
    editor.on('selectionUpdate', handleSelectionUpdate)
    // is changed to
    editor.on('transaction', handleTransactionUpdate)
    ```
Thanks!

## PR Checklist

<!-- Feel free to remove the lines that are not applicable or leave them unchecked. -->

-   [x] Pull request title follows the [Conventional Commits Specification](https://www.conventionalcommits.org/)
-   [ ] Added/updated unit test cases and/or end-to-end test cases
-   [ ] Added/updated documentation to Storybook or `README.md`

## Test plan
Doesn't change any existing features, just introduces fix for #259.
<!-- Please add a complete step-by-step description on how to test the changes. -->

<!--
- Open the preview Storybook deployed to Netlify
- Open the `Rich-text → Default` story
- Do something else…
    - [ ] Observe that this happened
-->

## Demo

| Before                      | After                      |
| --------------------------- | -------------------------- |
| Keyboard events and editor button clicks did not update the toolbar | Keyboard shortcuts and toolbar clicks correctly re-render editor toolbar.|
| ![image](https://github.com/Doist/typist/assets/36472216/54c17c0c-86fe-4908-bd37-ca0a26163e64) | ![image](https://github.com/Doist/typist/assets/36472216/0844ad99-1522-4140-a97c-7d1baadada9a)|

rfgamaral commented on stories/typist-editor/decorators/typist-editor-decorator/typist-editor-toolbar.tsx:
```diff
@@ -45,16 +45,16 @@ function TypistEditorToolbar({ editor }: TypistEditorToolbarProps) {
         function initializeEventListeners() {
-            function handleSelectionUpdate() {
+            function handleTransactionUpdate() {
                 // Force a rerender for every selection update in the editor - an event that ocurrs
                 // outside of the React lifecycle - to update the toolbar buttons `pressed` state
```
```suggestion
                // Force a rerender for every transaction in the editor - an event that ocurrs
                // outside of the React lifecycle - to update the toolbar buttons `pressed` state
```

rfgamaral reviewed (changes requested):
Thank you for fixing this @ayushanand18 🤗

It works as expected, but it's just missing a small change in the comment to match the `selectionUpdate` to `transaction` change. I'll merge it afterwards :)

rfgamaral commented:
@ayushanand18 FYI, I changed the PR title from `fix` to `chore` because `fix` would publish a new version (which we don't need for this one), removed the issue number from the title (which is added automatically to the commit message once the PR is merged), and also re-worded the commit message a bit to make it more clear what it's actually doing.

ayushanand18 commented:
> FYI, I changed the PR title from `fix` to `chore` because `fix` would publish a new version [...], removed the issue number from the title [...], and also re-worded the commit message a bit [...]

Thanks a lot @rfgamaral! I'll make sure to follow those guidelines for any future contributions.
